projects/valtimo/task/src/lib/components/task-detail-modal/task-detail-modal.component.html

```diff
@@ -13,16 +13,42 @@
   ~ See the License for the specific language governing permissions and
   ~ limitations under the License.
   -->
-
-<valtimo-modal
-  #taskDetailModal
-  elementId="taskDetailModal"
-  [title]="(page$ | async)?.title"
-  [subtitle]="(page$ | async)?.subtitle"
-  [templateBelowSubtitle]="assignUserToTask"
->
+<cds-modal #taskDetailModal id="taskDetailModal">
+  <cds-modal-header (closeSelect)="modal.open = false">
+    <div class="cds-modal-header">
+      <div class="title-container">
+        <p class="cds--modal-header__heading cds--type-beta">{{ (page$ | async)?.title }}</p>
+        <p class="cds--modal-header__label cds--type-delta">{{ (page$ | async)?.subtitle }}</p>
+      </div>
+      <div *ngIf="intermediateSaveEnabled">
+        <cds-tooltip
+          [caret]="true"
+          [description]="'formManagement.intermediateSave.save' | translate"
+        >
+          <svg
+            class="cds--btn__icon save-icon-margin cursor-pointer"
+            cdsIcon="save"
+            size="16"
+            (click)="saveCurrentProgress()"
+          ></svg>
+        </cds-tooltip>
+        <cds-tooltip
+          [caret]="true"
+          [description]="'formManagement.intermediateSave.clear' | translate"
+        >
+          <svg
+            class="cds--btn__icon cursor-pointer"
+            cdsIcon="trash-can"
+            size="16"
+            (click)="showConfirmationModal$.next(true)"
+          ></svg>
+        </cds-tooltip>
+      </div>
+    </div>
+  </cds-modal-header>
   <div
     body
+    class="cds--modal-content pb-1"
     *ngIf="{
```

Comment:
You can add this whole *ngIf on the cds-modal and then you won't have to consume page$ with async in the header